src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -742,14 +754,34 @@ public async Task CommitAsync(bool previewChanges)
     {
         if (this.RenameService.GlobalOptions.GetOption(InlineRenameSessionOptionsStorage.RenameAsynchronously))
         {
-            await CommitWorkerAsync(previewChanges, canUseBackgroundWorkIndicator: true).ConfigureAwait(false);
+            await StartCommitAsync(previewChanges, canUseBackgroundWorkIndicator: true).ConfigureAwait(false);
         }
         else
         {
             CommitSynchronously(previewChanges);
         }
     }
 
+    private async Task<bool> StartCommitAsync(bool previewChanges, bool canUseBackgroundWorkIndicator)
+    {
+        await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+        // We are going to commit in async manner.
+        // 1. If the session is dismissed, stop and do not start another commit task
+        if (_dismissed)
+        {
+            return false;
+        }
+
+        if (_commitTask is null || !IsCommitInProgress)
```

Comment:
how can we have a commit in progress if we're calling into StartCommit?  doc?

Reply:
It's possible and I will add this to the comment.

Reply:
done. Doc added.

Reply:
Added.

Reply:
Oh a unexpected case just comes to my mind. Mark this as unresovled.